tests/PactNet.Native.Tests/NativeRequestBuilderTests.cs

```diff
@@ -93,6 +93,14 @@ public void WithQuery_RepeatedQuery_SetsIndex()
             this.mockServer.Verify(s => s.WithQueryParameter(this.handle, "other", "value", 0));
         }
 
+        [Fact]
+        public void WithHeader_WhenCalledWithMatcher_AddsSerialisedHeaderParam()
+        {
+            this.builder.WithHeader("name", PactNet.Matchers.Match.Regex("header", "^header$"));
+
+            this.mockServer.Verify(s => s.WithRequestHeader(this.handle, "name", "{\"pact:matcher:type\":\"regex\",\"value\":\"header\",\"regex\":\"^header$\"}", 0));
```

Comment:
I think the JSON probably needs splitting out into an `expected` var so it's easier to read given the long line it creates.

Reply:
👍